WebRoot/js/zimbraMail/calendar/view/ZmCalColView.js

```diff
@@ -2842,11 +2842,11 @@ function(ev) {
         var col = data.view._getColFromX(data.gridX);
 		Dwt.setVisible(data.newApptDivEl, false);
 		if (data.isAllDay) {
-			appCtxt.getCurrentController().newAllDayAppointmentHelper(data.startDate, data.endDate, null, mouseEv.shiftKey);
+			(col && col.cal) ? appCtxt.getCurrentController().newAllDayAppointmentHelper(data.startDate, data.endDate, col.cal.id, mouseEv.shiftKey) : appCtxt.getCurrentController().newAllDayAppointmentHelper(data.startDate, data.endDate, null, mouseEv.shiftKey)
 		} else {
 			var duration = (data.endDate.getTime() - data.startDate.getTime());
 			if (duration < ZmCalColView._MSEC_DURATION) duration = ZmCalColView._MSEC_DURATION;
-			appCtxt.getCurrentController().newAppointmentHelper(data.startDate, duration, null, mouseEv.shiftKey);
+			(col && col.cal) ? appCtxt.getCurrentController().newAppointmentHelper(data.startDate, duration, col.cal.id, mouseEv.shiftKey) : appCtxt.getCurrentController().newAppointmentHelper(data.startDate, duration, null, mouseEv.shiftKey)
```

Comment:
I think it can be simplified like this:
```
var col = data.view._getColFromX(data.gridX);
var folderId = col && col.cal && col.cal.id;
Dwt.setVisible(data.newApptDivEl, false);
if (data.isAllDay) {
	appCtxt.getCurrentController().newAllDayAppointmentHelper(data.startDate, data.endDate, folderId, mouseEv.shiftKey);
} else {
	//...
	appCtxt.getCurrentController().newAppointmentHelper(data.startDate, duration, folderId, mouseEv.shiftKey);
}
```

Reply:
Will this work ?

`appCtxt.getCurrentController().newAppointmentHelper(data.startDate, duration, col && col.cal ? col.cal.id : null, mouseEv.shiftKey);`

Reply:
I think it's less readable...

Reply:
I have implemented both the suggested solutions and they work fine. Please let me know which implementation I should go for.

Reply:
I am fine with both. If my code is less readable than skip it.

Reply:
Okay. Please pick the first one.